### ISSUE TITLE
Triple dash --- in markdown YAML header makes all file to be skipped from check

### ISSUE THREAD

**Issue by nikcleju** [valentjn/vscode-ltex#104]
**Describe the bug**
I try to spell check a markdown file starting with a Pandoc YAML header (delimted by ---), it doesn't work.
It seems the triple dash is delimiter of the YAML header block is the problem, with 1 or 2 dashes it works, but Pandoc needs all 3.

**Steps to reproduce**
Clean install of latest version (6.3.0), default settings.

**Expected behavior**
Spell check :)

**Sample document**
If the bug occurs for a specific document (e.g. LaTeX), please paste it here. If your document is very long or confidential, please create and attach a smaller example for which the bug still occurs so that we can reproduce it.

<details>

```markdown
---
# General Pandoc and pacdoc-citeproc options
title: Title 
---

# Introduction

Hellook
```

</details>

**LTeX configuration**
Please paste all configuration settings starting with `ltex.` from your `settings.json`. You can help us by temporarily removing some irrelevant settings from your `settings.json` and see if the bug still occurs.

<details>

```
{
  "enabled": true,
  "language": "en-US",
  "dictionary": {},
  "disabledRules": {},
  "enabledRules": {},
  "ltex-ls": {
    "path": "",
    "languageToolHttpServerUri": ""
  },
  "java": {
    "path": "",
    "initialHeapSize": 64,
    "maximumHeapSize": 512
  },
  "commands": {
    "ignore": [],
    "dummy": []
  },
  "environments": {
    "ignore": []
  },
  "markdown": {
    "ignore": [
      "CodeBlock",
      "FencedCodeBlock",
      "IndentedCodeBlock"
    ],
    "dummy": [
      "AutoLink",
      "Code"
    ]
  },
  "ignoreRuleInSentence": [],
  "configurationTarget": {
    "addToDictionary": "global",
    "disableRule": "workspaceFolder",
    "ignoreRuleInSentence": "workspaceFolder"
  },
  "additionalRules": {
    "motherTongue": "",
    "languageModel": "",
    "neuralNetworkModel": "",
    "word2VecModel": ""
  },
  "sentenceCacheSize": 2000,
  "diagnosticSeverity": "error",
  "clearDiagnosticsWhenClosingFile": true,
  "trace": {
    "server": "off"
  },
  "javaHome": null,
  "performance": {
    "initialJavaHeapSize": null,
    "maximumJavaHeapSize": null,
    "sentenceCacheSize": null
  }
}
```

</details>

**"LTeX Language Server" log file**
First, reproduce the bug. Then, go to `View` → `Output` and select `LTeX Language Server` in the drop-down list. Paste this log here:

<details>

```
[... truncated]
Aug 25, 2020 11:20:38 PM org.bsplines.ltexls.DocumentChecker checkAnnotatedTextFragment
INFO: Checking the following text in language 'en-US' via LanguageTool: '\n\n\n\n\n\n\n\n\n\n\n\n'
Aug 25, 2020 11:20:38 PM org.bsplines.ltexls.DocumentChecker checkAnnotatedTextFragment
INFO: Obtained 0 rule matches
Aug 25, 2020 11:20:38 PM org.bsplines.ltexls.DocumentChecker checkAnnotatedTextFragment
INFO: Checking the following text in language 'en-US' via LanguageTool: '\n\n\n\n\n\n\n\n\n\n\n\n'
Aug 25, 2020 11:20:38 PM org.bsplines.ltexls.DocumentChecker checkAnnotatedTextFragment
INFO: Obtained 0 rule matches
Aug 25, 2020 11:20:43 PM org.bsplines.ltexls.DocumentChecker checkAnnotatedTextFragment
INFO: Checking the following text in language 'en-US' via LanguageTool: '\n\n\n\n\n\n\n\n\n\n\n'
Aug 25, 2020 11:20:43 PM org.bsplines.ltexls.DocumentChecker checkAnnotatedTextFragment
INFO: Obtained 0 rule matches

```

</details>

**"LTeX Language Client" log file**
First, set the `ltex.trace.server` setting in your `settings.json` to `"verbose"`. Then, reload the VS Code window and reproduce the bug. Go to `View` → `Output` and select `LTeX Language Client` in the drop-down list. Paste this log here (note: it will contain your checked document):

<details>

```
2020-08-25T20:12:45.377Z Info: Setting LTeX UI language to 'en-us'.
2020-08-25T20:12:45.378Z Info: Loading i18n messages...
2020-08-25T20:12:45.380Z Info: Loading default i18n messages...
2020-08-25T20:12:45.383Z Info: 
2020-08-25T20:12:45.384Z Info: ltex.ltex-ls.path not set.
2020-08-25T20:12:45.384Z Info: Searching for ltex-ls in 'c:\Users\uidj6654\.vscode\extensions\valentjn.vscode-ltex-6.3.0\lib'...
2020-08-25T20:12:45.386Z Info: ltex-ls found in 'c:\Users\uidj6654\.vscode\extensions\valentjn.vscode-ltex-6.3.0\lib\ltex-ls-6.3.0'.
2020-08-25T20:12:45.386Z Info: 
2020-08-25T20:12:45.386Z Info: ltex.java.path not set.
2020-08-25T20:12:45.386Z Info: Using ltex-ls from 'c:\Users\uidj6654\.vscode\extensions\valentjn.vscode-ltex-6.3.0\lib\ltex-ls-6.3.0'.
2020-08-25T20:12:45.386Z Info: Using Java from PATH or JAVA_HOME (may fail if not installed).
2020-08-25T20:12:45.391Z Info: Testing ltex-ls...
2020-08-25T20:12:45.391Z Info:   Command: "c:\\Users\\uidj6654\\.vscode\\extensions\\valentjn.vscode-ltex-6.3.0\\lib\\ltex-ls-6.3.0\\bin\\ltex-ls.bat"
2020-08-25T20:12:45.391Z Info:   Arguments: ["--version"]
2020-08-25T20:12:45.391Z Info:   env['JAVA_HOME']: undefined
2020-08-25T20:12:45.391Z Info:   env['JAVA_OPTS']: "-Xms64m -Xmx512m"
2020-08-25T20:12:46.978Z Info: Test failed due to too old Java. You have Java 8 (version 1.8.0_261). Please upgrade to Java 11 or newer.
2020-08-25T20:12:46.978Z Info: 
2020-08-25T20:12:46.978Z Info: Searching for Java in 'c:\Users\uidj6654\.vscode\extensions\valentjn.vscode-ltex-6.3.0\lib'...
2020-08-25T20:12:46.978Z Info: Java found in 'c:\Users\uidj6654\.vscode\extensions\valentjn.vscode-ltex-6.3.0\lib\jdk-11.0.8+10-jre'.
2020-08-25T20:12:46.978Z Info: Using ltex-ls from 'c:\Users\uidj6654\.vscode\extensions\valentjn.vscode-ltex-6.3.0\lib\ltex-ls-6.3.0'.
2020-08-25T20:12:46.978Z Info: Using Java from 'c:\Users\uidj6654\.vscode\extensions\valentjn.vscode-ltex-6.3.0\lib\jdk-11.0.8+10-jre'.
2020-08-25T20:12:46.980Z Info: Testing ltex-ls...
2020-08-25T20:12:46.980Z Info:   Command: "c:\\Users\\uidj6654\\.vscode\\extensions\\valentjn.vscode-ltex-6.3.0\\lib\\ltex-ls-6.3.0\\bin\\ltex-ls.bat"
2020-08-25T20:12:46.980Z Info:   Arguments: ["--version"]
2020-08-25T20:12:46.980Z Info:   env['JAVA_HOME']: "c:\\Users\\uidj6654\\.vscode\\extensions\\valentjn.vscode-ltex-6.3.0\\lib\\jdk-11.0.8+10-jre"
2020-08-25T20:12:46.980Z Info:   env['JAVA_OPTS']: "-Xms64m -Xmx512m"
2020-08-25T20:12:48.248Z Info: Test successful!
2020-08-25T20:12:48.248Z Info: 
2020-08-25T20:12:48.259Z Info: Starting ltex-ls...
2020-08-25T20:12:48.259Z Info:   Command: "c:\\Users\\uidj6654\\.vscode\\extensions\\valentjn.vscode-ltex-6.3.0\\lib\\ltex-ls-6.3.0\\bin\\ltex-ls.bat"
2020-08-25T20:12:48.259Z Info:   Arguments: []
2020-08-25T20:12:48.259Z Info:   env['JAVA_HOME']: "c:\\Users\\uidj6654\\.vscode\\extensions\\valentjn.vscode-ltex-6.3.0\\lib\\jdk-11.0.8+10-jre"
2020-08-25T20:12:48.259Z Info:   env['JAVA_OPTS']: "-Xms64m -Xmx512m"
2020-08-25T20:12:48.259Z Info: 
2020-08-25T20:22:11.429Z Info: Creating bug report...

```

</details>

**Version information**
List here the version information of the relevant software.

- Operating system: Windows_NT (win32), x64, 10.0.17763
- VS Code: 1.48.2
- vscode-ltex: 6.3.0
- ltex-ls: 6.3.0
- Java: 11.0.8

**Additional context/information**
You can add any other context or information about the problem here.

**Comment by valentjn** [comment]
I can reproduce this. Support for skipping YAML front matter was added in 6.2.0, there's even a [test](https://github.com/valentjn/ltex-ls/blob/3060e582006d3ea5d474211fc6eb0814b518f5a6/ltexls-core/src/test/java/org/bsplines/ltexls/parsing/markdown/MarkdownAnnotatedTextBuilderTest.java#L53) for it. I think the problem is that the whole document, i.e., including the front matter, is parsed by the Markdown parser, and that obviously doesn't work. The front matter has to be removed before passing the document to the Markdown parser.

I'm not sure what the exact conditions are for the bug to occur, but as a workaround try inserting an empty line before the second `---`, that seems to help (or downgrade to 6.1.1).

**Comment by valentjn** [comment]
Fixed in 7.0.0.
